src/OrbitGl/TimelineUiTest.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include "GlCanvas.h"
+#include "MockBatcher.h"
+#include "MockTextRenderer.h"
+#include "MockTimelineInfo.h"
+#include "TimelineUi.h"
+#include "Viewport.h"
+
+namespace orbit_gl {
+
+class TimelineUiTest : public TimelineUi {
+ public:
+  TimelineUiTest(int width)
+      : TimelineUi(nullptr /*parent*/, &mock_timeline_info, &viewport_, &layout_),
```

Comment:
You can't do that. `mock_timeline_info`, `viewport_`, and `layout_` have not yet been initialized when the `TimelineUi` constructor is called. You shouldn't assume that the constructor of `TimelineUi` does not derefence the pointers.

Reply:
Could you make `TimelineUi` a member variable instead of a base class?

Reply:
I'm trying to test protected methods (UpdatePrimitives & Draw). This methods should be public in the future, I think, but for now this is the solution that we found to test that methods.

Reply:
Then you need to make `mock_timeline_info`, `viewport_`, and `layout_` arguments to the `TimelineUiTest` constructor. They unfortunately can't live in this class.

Reply:
Better now?